tests/api/getWilayaByBaladyiaName.test.js

```diff
@@ -0,0 +1,183 @@
+            name_en: "HASSI GARA",
+            name_ar: "حاسي القارة"
+          }
+        ]
+      },
+      {
+        code: 4703,
+        name: "DHAYET BEN DHAHOUA",
+        name_ar: "ضاية بن ضحوة",
+        name_en: "DHAYET BEN DHAHOUA",
+        baladyiats: [
+          {
+            code: 4703,
+            name: "DHAYET BENDHAHOUA",
+            name_en: "DHAYET BENDHAHOUA",
+            name_ar: "ضاية بن ضحوة"
+          }
+        ]
+      },
+      {
+        code: 4704,
+        name: "BERRIANE",
+        name_ar: "بريان",
+        name_en: "BERRIANE",
+        baladyiats: [
+          {
+            code: 4704,
+            name: "BERRIANE",
+            name_en: "BERRIANE",
+            name_ar: "بريان"
+          }
+        ]
+      }]
+  }];
+
+  const noBaladyiaMockData = [
+    {
+      mattricule: 12,
+      dairats: [
+        {
+          code: 1090,
+          name: "TIMIMOUN (wilaya déléguée)",
+          name_ar: "تيميمون (ولاية منتدبة)",
+          name_en: "TIMIMOUN (wilaya déléguée)"
+        },
+        {
+          code: 1250,
+          name: "BORDJ BADJI MOKHTAR (wilaya déléguée)",
+          name_ar: "برج باجي مختار (ولاية منتدبة)",
+          name_en: "BORDJ BADJI MOKHTAR (wilaya déléguée)",
+          baladyiats: [
+            {
+              code: 4702,
+              name: "EL MENIAA",
+              name_en: "EL MENIAA",
+              name_ar: "المنيعة"
+            },
+          ]
+        }
+      ]
+    }
+  ];
+
+  let getWilayaByBaladyiaName;
+
+  beforeEach(()=> {
+    getWilayaByBaladyiaName = require('../../src/api/getWilayaByBaladyiaName');
+  });
+
+  it('should export a function', () => {
+    expect(typeof getWilayaByBaladyiaName).toBe('function');
+  });
+
+  it('should return a curried function that returns the data', () => {
+    const fn = getWilayaByBaladyiaName(mockData);
+
+    expect(typeof fn).toBe('function');
+  });
+
+  it('should return null if the wilaya with the given baladyia is not found', () => {
+    const result = getWilayaByBaladyiaName(mockData)("foo");
+    expect(result).toBeNull();
+  });
+  it('should return null if the wilaya with the given baladyia invalid name', () => {
+    const result = getWilayaByBaladyiaName(mockData)("fo");
+
+    expect(result).toBeNull();
+  });
+
+  it('should return null if there\'s no daira', () => {
+    const result = getWilayaByBaladyiaName(mockData)();
+
+    expect(result).toBeNull();
+  });
+
+  it('should return null if the wilaya with the given daira is empty', () => {
+    const result = getWilayaByBaladyiaName(mockData)("ZRZR");
+
+    expect(result).toBeNull();
+  });
+
+  it('should return matching wilaya object with arabic name', () => {
+    const result = getWilayaByBaladyiaName(mockData)("GHARDAIA", ["mattricule", "name"]);
+
+    expect(result).toEqual({
+      mattricule: 47,
+      name: "Ghardaïa",
+    });
+  });
+
+  it('should return matching wilaya object with english name', () => {
```

Comment:
I'd suggest adding something like `EL MENIAA - EN` in the initial dataset to cover this branch, or we can remove this test case anyway